### PR TITLE
Feat: Activators, Layer Tally | Fix: SelectPlaylist

### LINKED ISSUE #41: Instance Feedback: Live Input in program/preview
I am not sure if this should be a modification to the existing feedback function or a separate type of feedback.  I would like to be able to set feedback on a button if any input on a multi view input is in the preview/program.  I believe the tally natively works that way so maybe that could be how a new function works.  Tally is Program (set input name) and Tally is Preview (set input name).  Thanks for reading!

### LINKED ISSUE #53: Feat: Activator support
Subscribing to activator events over the TCP connection.

This will allow access to more data that isn't available through the HTTP API (such as call audio source).
- More feedbacks not previously available
- More responsive feedback
- Reduced need to poll the API as frequently for some users
- Reduced need to update feedback/variables as often, potentially leading to Companion performance improvements

### LINKED ISSUE #58: Function Playlist - Open Playlist not working
It seems that the above mentioned vMix function is not working.
I created a playlist in vMix with a name like "Intro". I tried to open this playlist by using the "Playlist - Open Playlist" function in Companion V2.1.0 and entering the name of the playlist in the options field. Unfortunately it doesn't work.

Using "Run custom command" with command "SelectPlayList Value=Intro" instead, works.

Best regards,

René

### PR DESCRIPTION
**Pull request by thedist**
**Added activator support by subscribing to activator messages on the TCP connection.**

Several messages are currently unused as there's either no feedback that requires them, or require further updates in vMix v24 before they are of use.

Activators that do change feedback, or variables, (such as prv/prgm feedback, or volume variables) are buffered and set to update no faster than once every 50ms. This is fine in testing, and improves the responsiveness of feedback greatly and in some situations can allow for slower API polling without impact on feedback responsiveness. Activator support has also reduced the number of unnecessary `checkFeedbacks` and `setVariable` calls.

**Layer Tally Indicators**
Prv/Prgm feedback now has a new dropdown box allowing the choice of none, border, or corner, indicators. These use the same colour as the feedback bg colour, and will trigger whenever an input is not in preview/program (in which case the solid background is used) but is either a multiview layer, or an overlay.

**Minor bug fix**

Fixes #58 
resolves #41 
resolves #53

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -807,27 +807,53 @@ exports.getActions = function() {
...
-			label: 'Replay - Replay A Camera',
+			label: 'Replay - A Camera',
 			options: [
 				{
 					type: 'dropdown',
 					label: 'Camera',
 					id: 'functionID',
 					default: 'ReplayACamera1',
-					choices: ['ReplayACamera1', 'ReplayACamera2', 'ReplayACamera3', 'ReplayACamera4'].map((item, index) => ({ id: item, label: `Camera ${index + 1}` }))
+					choices: [1, 2, 3, 4, 5, 6, 7, 8].map((item) => ({ id: `ReplayACamera${item}`, label: `Camera ${item}` }))
```
Comment (krocheck) [inline src/actions.js:817]: this is a breaking change needing an upgrade script

File: src/actions.js [hunk 2]
```diff
@@ -807,27 +807,53 @@ exports.getActions = function() {
...
-			label: 'Replay - Replay B Camera',
+			label: 'Replay - B Camera',
 			options: [
 				{
 					type: 'dropdown',
 					label: 'Camera',
 					id: 'functionID',
 					default: 'ReplayBCamera1',
-					choices: ['ReplayBCamera1', 'ReplayBCamera2', 'ReplayBCamera3', 'ReplayBCamera4'].map((item, index) => ({ id: item, label: `Camera ${index + 1}` }))
+					choices: [1, 2, 3, 4, 5, 6, 7, 8].map((item) => ({ id: `ReplayBCamera${item}`, label: `Camera ${item}` }))
```
Comment (krocheck) [inline src/actions.js:830]: ditto

File: src/feedback.js [hunk 3]
```diff
@@ -451,8 +504,12 @@ exports.executeFeedback = function(feedback, bank) {
 		if (idCheck || titleCheck || guidCheck) {
 			return { color: feedback.options.fg, bgcolor: feedback.options.bg };
 		}
+
+		if (feedback.options.tally !== undefined && feedback.options.tally !== '' && this.data.mix[mix][type + 'Tally'].includes(input.key)) {
```
Comment (krocheck) [inline src/feedback.js:508]: I'm not sure your return schemes make sense here ... shouldn't there be and the `if` as shown then an `else` or `else if` for normal bg + fg colors?  Makes no sense you return those in the earlier `if`, leading me to believe in most cases this `png64` would never be returned.

**Comment by thedist** [comment]
I'm going to close this Pull Request, refactor the new features into separate branches so that the changes that can be added now can be, and the layer indicators that require Companions 'develop' branch can wait until the develop branch is merged into master.

**Pull request closed without merging**